test/Makefile.am

```diff
@@ -70,7 +70,7 @@ TEST_PROG= testhdf5 \
            enc_dec_plist_cross_platform getname vfd ros3 s3comms hdfs ntypes \
            dangle dtransform reserved cross_read freespace mf vds file_image \
            unregister cache_logging cork swmr thread_id vol timer event_set \
-           onion mt_id_test
+           onion mt_id_test threads/testmthdf5
```

Comment:
It gets ugly quickly due to boilerplate, but we should follow the traditional pattern of having a Makefile.am in `threads/`, `threads/unit`, etc. to handle the respective files in each directory.